### PR TITLE
fix: add missing http filters to the http filter chain

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR patches HTTP filters on the existing `HCM` of an HTTP listener's `DefaultFilterChain`.

Fix https://github.com/envoyproxy/gateway/issues/2742

### CODE REVIEW

File: internal/xds/translator/testdata/out/xds-ir/http3.listeners.yaml [hunk 1]
```diff
@@ -2,21 +2,18 @@
     socketAddress:
       address: 0.0.0.0
       portValue: 10443
-      protocol: UDP
```
Comment (arkodg) [inline internal/xds/translator/testdata/out/xds-ir/http3.listeners.yaml:5]: why did this yaml change ?

**Comment by arkodg** [comment]
hey @zhaohuabing thanks for working on the fix + the refactor
can we split up the refactor into another follow PR ? It allows us to back port this into v1.0.0

File: internal/xds/translator/testdata/out/xds-ir/http3.listeners.yaml (hunk 1 above)
Comment (zhaohuabing) [reply internal/xds/translator/testdata/out/xds-ir/http3.listeners.yaml:5]: The order of the quick and tcp listener changes.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**